src/rules/no_const_assign.rs

```diff
@@ -59,66 +57,70 @@ impl LintRule for NoConstAssign {
   }
 }
 
-struct NoConstAssignVisitor<'c, 'view> {
-  context: &'c mut Context<'view>,
-}
-
-impl<'c, 'view> NoConstAssignVisitor<'c, 'view> {
-  fn new(context: &'c mut Context<'view>) -> Self {
-    Self { context }
-  }
+struct NoConstAssignHandler;
 
-  fn check_pat(&mut self, pat: &Pat, span: Span) {
+impl NoConstAssignHandler {
+  fn check_pat(&mut self, pat: &Pat, span: Span, ctx: &mut Context) {
```

Comment:
Now these `check_*` methods can be changed to take `&self` instead of `&mut self`, right? If possible, it would be preferable.
Or it even looks like we could write these methods as normal functions rather than methods, since they don't depend on `self`. It's a matter of preference, but I personally prefer to have them as normal functions.